### ISSUE TITLE
Feature Request: Detect changed prices

### ISSUE THREAD

**Issue by dekiesel** [orangecoding/fredy#102]
**Is your feature request related to a problem? Please describe.**
-

**Describe the solution you'd like**
There was a house I was interested in, but it had a damaged basement. Fredy found the listing since it was in my price range. I decided to pass on the house because `price+basement repair>budget`. 
Today I saw that they dropped the price by 100k and suddenly the house was within my budget, but I didn't see the new price until after it was sold. 

tl;dr: I'd like to get a notification when a listing has a price change.

**Describe alternatives you've considered**
Manually checking every listing, but that kinda defeats the purpose of fredy.

**Comment by orangecoding** [comment]
Uhh.. cool feature. The problem of this is I don't store the actual data of a listing I sent, but rather only the listing id to make sure I don't send listings twice. For this to work, I would need to store specific props like price...

Are there ANY other props you could think of that might justify sending a listing again?

**Comment by dekiesel** [comment]
Hmmm, that's a great question. Maybe square meters (in case there was a typo or something)? But I doubt that happens often if ever. Or if the realtor cut goes up or down?

It might be an idea to just save everything (that's not deemed a duplicate listing) if you touch the code anyway so that it is easier to get alerts for any changes in case somebody comes up with a good use case

**Comment by orangecoding** [comment]
I think I have a better idea. Rather than storing the listing id, I could generate a hash that includes the listing id and props like the price. If any of which changes, the hash changes, thus the listing is being sent again.

Wdyt?

**Comment by dekiesel** [comment]
May I ask why you'd rather not store the ID? I'd personally like it if all the details were saved because I wanted to do some analytics with the data anyway until I realized that not everything gets saved.

The hash would definitely work though. Only issue is see is that it only indicates that *something* has changed. In case somebody else comes around with a different use case the hash won't be enough.

**Comment by orangecoding** [comment]
You are right, it only indicates something has changed, not what has changed. The reason I don't want to store everything is - it is a lot of data (most of people I know of running Fredy on a Pi)
- it changes the structure of how I store data internally and I do not have the capacity at the moment to do those big changes...

What kind of analytics do you want to do?

**Comment by dekiesel** [comment]
I don't think it's a lot of data, not even for a pi. You might not be able to do analytics on it (depending on the query you want to run), but I'd be surprised if a pi couldn't easily save this amount of data per listing.

I wanted to see whether there was a change of average pricing over the past weeks/months and if so what the rate of change was. Now that interest is high I wanted to see whether that has an effect on the houses in the area I am searching in.

I work as a data/BI engineer, but I haven't used javascript in 20 years so unfortunately I can't help with PRs :/

**Comment by orangecoding** [comment]
Mhm. Well I don't say it wouldn't be a nice feature, I just don't think it is worth storing it with the scraped id's as your use case is pretty unique. At least nobody ever asked for something even close to this.

I truly think (especially for data science) it is better to store the values somewhat differently.. maybe even in another database...? 🤔

**Comment by dekiesel** [comment]
I am happy with just the hash solution. The most important thing is that I get to know *something* happened to the listing. Being able to do analytics would just be the cherry on top.

**Comment by dekiesel** [comment]
The DB seems to be a json file, right? Would you be willing to just dump the data you scraped for new/changed listings into a separate file? That way you wouldn't need to take data types into account and people could still do their analytics.

**Comment by orangecoding** [comment]
I can do that, but that is definitely a bigger feature and I'd turn it off by default as very little group of ppl would be interested in this

**Comment by orangecoding** [comment]
Hey @dekiesel my latest change introduce a hash including id and price. Please check it out. 

If it works, please close the issue.

**Comment by dekiesel** [comment]
Thanks! That was quick! I'll give it a spin. I'll just have to wait for somebody to update a price.

**Comment by orangecoding** [comment]
Hey. I tested it myself and can confirm it works. I'll therefor close this issue

**Comment by dekiesel** [comment]
I just noticed that there might be a slight issue due to this change. Duplicate detection seems to have taken a hit.

I now get duplicate listings from at least einsAImmobilien where the only difference is the monthly price, which is influenced by the interest rate that the site uses to calculate the monthly price.

Example:

`775.700,00 EUR Kaufpreis ab 2.475 € mtl.*`

vs

`775.700,00 EUR Kaufpreis ab 2.305 € mtl.* `

These are for the same listing, one day apart.

**Comment by dekiesel** [comment]
@orangecoding 

I am not fluent in javascript, but I do a bit of scraping with selenium/python.

It seems that in the config [here](https://github.com/orangecoding/fredy/blob/master/lib/provider/einsAImmobilien.js) in line 26 the whole div is scraped, but checking einsaimmobilien I see that only the `text` should get scraped:

![image](https://github.com/user-attachments/assets/4470077d-5c14-465e-9d9d-c0eeefb67b69)

**Comment by orangecoding** [comment]
what you could do is check for (lower case) eur. and if it is available, remove everything after and take this as the result.

**Comment by dekiesel** [comment]
Doesn't js have the option to just get the ``#text`? Seems more robust than just cutting somewhere that might change (eg EUR to €).
